### PR TITLE
Use storage backend from VscStorage

### PR DESCRIPTION
**Pull request by lexming**
In Brussels, we now have different storage backends for the different VSC filesystems. VSC_HOME and VSC_DATA are using an off-cluster storage from Huawei called OceanStor, and VSC_SCRATCH is using the usual GPFS filesystem. Therefore, we enhanced `VscStorage` to support different backends for each filesystem described in `STORAGE_CONFIGURATION_FILE`.

This PR updates vsc-filesystems-quota to make it agnostic of the storage backend used for each filesystem.

Depends on:
* https://github.com/hpcugent/vsc-config/pull/177
* https://github.com/hpcugent/vsc-filesystems/pull/83

The symbol :warning: denotes breaking changes. I only introduced breaking changes on internal methods and to `get_mmrepquota_maps`, which is not used anywhere else AFAIK, so the damage should be very minimal.

Changelog:
* replace `GpfsOperations` with corresponding operator loaded for the corresponding backend in`VscStorage`
* :warning: replace `get_mmrepquota_maps` with storage agnostic method `get_quota_maps`
* :warning: arguments of `_update_quota_entity` changed to rely on the `VscStorage` and its backend operator
* move code specific to GPFS to `vsc.filesystems.gpfs.GpfsOperations`: those are now used through `fs_backend.determine_grace_period`, `fs_backend.get_fileset_name`, `fs_backend.get_quota_fileset` and `fs_backend.get_quota_owner`
* remove `QuotaException` as it is not used after the move of code to `GpfsOperations`
* :warning: move `InodeLog` from vsc.filesystem.quota.tools to vsc.filesystem.quota.inodes

### CODE REVIEW

**Comment by lexming** [comment]
Tests are positive with the required updates on its dependencies:
https://gist.github.com/c25c721e48060a186b7807ef8e5fd6dc

File: lib/vsc/filesystem/quota/tools.py [hunk 1]
```diff
@@ -234,104 +224,89 @@ def process_user_quota(storage, gpfs, storage_name, filesystem, quota_map, user_
...
-    # Iterate over a list of named tuples -- GpfsQuota
-    for (user, gpfs_quota) in quota_map['USR'].items():
-        user_quota = user_map.get(user, QuotaUser(storage, filesystem, user))
+    filesystem = storage[storage_name].filesystem
+
+    fs_backend = load_storage_operator(storage[storage_name])
+    quotas = fs_backend.list_quota(devices=filesystem)
+    quota_map = quotas[filesystem]
+
+    logging.info("ordering USR quota for storage %s", storage_name)
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py]: `USR` -> some enum.value

File: bin/dquota.py [hunk 2]
```diff
@@ -70,50 +70,41 @@ def main():
         client = AccountpageClient(token=opts.options.access_token)
 
         user_id_map = map_uids_to_names()
-        gpfs = GpfsOperations()
         storage = VscStorage()
```
Comment (stdweird) [inline bin/dquota.py:73]: best to make a `VscStorageOperator` class in vsc-filesystems that does eg
```
class VscStorageOperator(VscStorage):
   def __init__
      VscStorgae.__init__
      load_storage_operator(...)
  

```

File: lib/vsc/filesystem/quota/tools.py (hunk 1 above)
Comment (lexming) [reply lib/vsc/filesystem/quota/tools.py]: fixed in https://github.com/hpcugent/vsc-filesystems-quota/pull/19/commits/03e2f3573385020373e31c9fe977bac475c6d667

File: bin/dquota.py (hunk 2 above)
Comment (lexming) [reply bin/dquota.py:73]: all done and updated in https://github.com/hpcugent/vsc-filesystems-quota/pull/19/commits/9526d40bb0abbacd6c467af2c218057da9b76550

**Comment by lexming** [comment]
@stdweird PR updated with latest developments in vsc-config and vsc-filesystems.  All stuff that is specific to GPFS is now leveraged through `StorageOperator`.

I'm also moving `InodeLog` and `process_inodes_information` from vsc.filesystem.quota.tools to their own module vsc.filesystem.quota.inodes. They do not interact with anything else in `tools` and this allows to greatly simplify the imports in `tools`.

**Comment by itkovian** [comment]
I am going to merge and solve issues in the other PRs.

**Pull request merged**